### ISSUE TITLE
Special identifier to wrap cell in a string macro

### ISSUE THREAD

**Issue by MasonProtter** [JuliaLang/IJulia.jl#920]
I think that given that we don't have a way to hook into custom repl modes other than the ones hard coded into IJulia, it could be really nice to have a special syntax for wrapping the contents of a cell in a string macro. The idea would be that the string macros used in this way are analogous to custom repl modes, and the special syntax would allow us to use them in such a way that would not interfere with syntax highlighting and would require less typing.

For instance, if someone did `using PyCall`, they have access to the `py_str` macro so they could just have a codeblock that looks like 
```julia
%% py
import numpy as np

def sinpi(x):
    return np.sin(np.pi * x)

sinpi(1)
```
would be translated into
```julia
py"""
import numpy as np

def sinpi(x):
    return np.sin(np.pi * x)

sinpi(1)
"""
````
Alternatively to the `%%` syntax, maybe we could just handle a comment on the first line specially, i.e. `#mode mymode` would use the `mymode_str` macro? I'm not sure what the best syntax to use would be.

**Comment by stevengj** [comment]
I'm not sure why `%% py` is better than `py""" .... """`  other than saving a couple characters?  The latter has the advantage that it won't be formatted as Julia code.

**Comment by MasonProtter** [comment]
Yes, maybe PyCall is a poor example because Python's indentation style could cause headaches. 

Basically, my intention would be to use this for special repl-like modes. There are cases where one might want to use a string macro for a code block but still mostly adhere to Julia's indentation rules. One example being https://github.com/swadey/LispSyntax.jl, or something like PyCall but for Matlab. Another example would be a side-project I'm working on to support [Bra-Ket notation](https://en.wikipedia.org/wiki/Bra–ket_notation) in julia with a string macro.
